lib/buffer.js

```diff
@@ -272,30 +272,64 @@ Buffer.concat = function(list, length) {
 };
 
 
+function base64ByteLength(str, len) {
+  var bytes = len;
+
+  // Handle padding
+  if (str[len - 1] === '=')
+    bytes--;
+  if (len > 2 && str[len - 2] === '=')
+    bytes--;
+
+  // Base64 ratio: 3/4
+  return (bytes * 3) >>> 2;
+}
+
+
 function byteLength(string, encoding) {
-  if (typeof(string) !== 'string')
-    string = String(string);
+  if (typeof string !== 'string')
+    string = '' + string;
 
-  if (string.length === 0)
+  var len = string.length;
+  if (len === 0)
     return 0;
 
-  switch (encoding) {
-    case 'ascii':
-    case 'binary':
-    case 'raw':
-      return string.length;
+  // Use a for loop to avoid recursion
+  var loweredCase = false;
+  for (;;) {
+    switch (encoding) {
+      case 'ascii':
+      case 'binary':
+      // Deprecated
+      case 'raw':
+      case 'raws':
```

Comment:
FWIW, I think 'raw' and 'raws' have been deprecated since v0.2 or v0.3.  I've never seen them being used in the wild either.